test/test_charm.py

```diff
@@ -430,3 +431,49 @@ def test_containers(self):
         self.assertIsInstance(meta.containers['test2'], ContainerMeta)
         self.assertEqual(meta.containers['test1'].name, 'test1')
         self.assertEqual(meta.containers['test2'].name, 'test2')
+
+    def test_containers_storage(self):
+        meta = CharmMeta.from_yaml("""
+name: k8s-charm
+storage:
+  data:
+    type: filesystem
+    location: /test/storage
+  other:
+    type: filesystem
+    location: /test/other
+containers:
+  test1:
+    mounts:
+      - storage: data
+        location: /test/storage
+      - storage: other
+        location: /test/other
```

Comment:
Having the location here (workload) match exactly the location above (charm) might give false negatives.
Could also add a test for when `location` under `storage` is omitted (should become `/var/lib/juju/storage/<name>/<num>`)

Reply:
+1 on testing for the default case.